x-pack/plugins/fleet/server/routes/fleet_proxies/handler.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RequestHandler } from '@kbn/core/server';
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+import type { TypeOf } from '@kbn/config-schema';
+
+import {
+  listFleetProxies,
+  createFleetProxy,
+  deleteFleetProxy,
+  getFleetProxy,
+  updateFleetProxy,
+} from '../../services/fleet_proxies';
+import { defaultFleetErrorHandler } from '../../errors';
+import type {
+  GetOneFleetProxyRequestSchema,
+  PostFleetProxyRequestSchema,
+  PutFleetProxyRequestSchema,
+} from '../../types';
+
+export const postFleetProxyHandler: RequestHandler<
+  undefined,
+  undefined,
+  TypeOf<typeof PostFleetProxyRequestSchema.body>
+> = async (context, request, response) => {
+  const coreContext = await context.core;
+  const soClient = coreContext.savedObjects.client;
+  // const esClient = coreContext.elasticsearch.client.asInternalUser;
+  try {
+    const { id, ...data } = request.body;
+    const proxy = await createFleetProxy(soClient, { ...data, is_preconfigured: false }, { id });
+
+    const body = {
+      item: proxy,
+    };
+
+    return response.ok({ body });
+  } catch (error) {
+    return defaultFleetErrorHandler({ error, response });
+  }
+};
+
+export const putFleetProxyHandler: RequestHandler<
+  TypeOf<typeof PutFleetProxyRequestSchema.params>,
+  undefined,
+  TypeOf<typeof PutFleetProxyRequestSchema.body>
+> = async (context, request, response) => {
+  try {
+    const coreContext = await await context.core;
+    // const esClient = coreContext.elasticsearch.client.asInternalUser;
+    const soClient = coreContext.savedObjects.client;
+
+    const item = await updateFleetProxy(soClient, request.params.itemId, request.body);
+    const body = {
+      item,
+    };
+
+    // TODO bump policies on update
```

Comment:
do you plan to fix this TODO in another pr?

Reply:
Yes the whole policy generation is coming in a following PR